source/lib/login.ts

```diff
@@ -78,15 +76,16 @@ export async function performLogin(
 	})
 
 	let responseUrl = new URL(loginResponse.url)
-	let responseMessage = responseUrl.searchParams.get('message')
+	// URLSearchParams.get requires a polyfill in react native
+	let responseMessage = responseUrl.href.includes('error=')
 	if (responseMessage) {
 		throw new LoginFailedError(`Login failed: ${responseMessage}`)
 	}
 
 	return {username, password}
 }
 
-type QueryFnData = SharedWebCredentials
+type QueryFnData = false | SharedWebCredentials
```

Comment:
After an out of band conversation with Hawken, it looks like the other half of the magic is within the `QueryFnData` type

> If we add " | false" to that line, and put useCredentials back without the throw, what happens?
> 
> - TData and TError are the internal type names used by useQuery
> - TData is the type returned by queryFn, and TError is … the types that "error" can be in the result of useQuery (more or less, "what errors can this throw")
>
> Except that TData is actually the "type returned by select:", which defaults to the identity function. 
> 
> So if we pretend that we have TQueryReturn and TSelectReturn,
> 
> let query = useQuery({
>   queryFn: () => blah, // returns TQueryReturn
>   select: (data: TQueryReturn) // returns TSelectReturn
> 
> query.data: TSelectReturn
> 
> And TData is TSelectReturn
> 
> I hate how much type inference is happening in useCredentials & friends - it makes horrid errors

Nice! This last change makes the contract work.